apps/website/src/pages/forms/web-dev-challenge/hackathon.astro

```diff
@@ -0,0 +1,255 @@
+								</p>
+							</div>
+
+							<div class="field">
+								<div class="checkbox-group">
+									<input
+										type="checkbox"
+										name="tocAgreement"
+										id="tocAgreement"
+										class="input checkbox"
+										required
+									/>
+									<label for="tocAgreement">
+										I agree to the terms and conditions.
+									</label>
+								</div>
+							</div>
+							
+							<div class="field">
+								<p>Only check this box if you DO NOT want your app to be shared by the sponsor.</p>
+								<div class="checkbox-group">
+									<input
+										type="checkbox"
+										name="doNotShare"
+										id="doNotShare"
+										class="input checkbox"
+									/>
+									<label for="doNotShare">
+										I <strong>DO NOT</strong> want the sponsor to showcase my web app as an example of using their tool
+									</label>
+								</div>
+							</div>
+						</section>
+
+						<section class="form-section">
+							<button type="submit" class="button">Submit</button>
+
+							<input type="hidden" name="id" value={userDetails._id} />
+							<input type="hidden" name="username" value={user?.username} />
+						</section>
+					</form>
+				</div>
+			</section>
+		</SignedIn>
+
+		<SignedOut>
+			<SignIn fallbackRedirectUrl={Astro.url.toString()} />
+		</SignedOut>
+	</main>
+</Layout>
+
+<template id="link-template">
+	<div class="add-link">
+		<div>
+			<label for="link_label[]">Label</label>
+			<input
+				type="text"
+				id="link_label[]"
+				name="link_label[]"
+				data-type="link-input"
+				class="input link"
+				required
+			/>
+		</div>
+
+		<div>
+			<label for="link_url[]">URL</label>
+			<input
+				type="url"
+				id="link_url[]"
+				name="link_url[]"
+				data-type="link-input"
+				class="input link"
+				required
+			/>
+		</div>
+
+		<button class="remove-link">
+			<svg
+				xmlns="http://www.w3.org/2000/svg"
+				width="16"
+				height="16"
+				viewBox="0 0 24 24"
+				fill="none"
+				stroke="currentColor"
+				stroke-width="1.5"
+				stroke-linecap="round"
+				stroke-linejoin="round"
+				class="lucide lucide-trash"
+				><path d="M3 6h18"></path><path
+					d="M19 6v14c0 1-1 2-2 2H7c-1 0-2-1-2-2V6"></path><path
+					d="M8 6V4c0-1 1-2 2-2h4c1 0 2 1 2 2v2"></path></svg
+			>
+			<span class="sr-only">Delete Link</span>
+		</button>
+	</div>
+</template>
+
+<style>
+	main {
+		background: var(--black);
+		padding: 40px 5cqw 60px;
+	}
+
+	p {
+		margin-block-start: 16px;
+
+		&:first-child {
+			margin-block-start: 0;
+		}
+	}
+
+	.terms-conditions {
+		background: var(--bg);
+		border: 1px solid color-mix(in oklch, var(--text-muted) 25%, transparent);
+		border-radius: 3px;
+		display: block;
+		font-size: 0.875em;
+		margin-block-start: 16px;
+		max-block-size: 35dvb;
+		overflow-y: scroll;
+		padding: 8px;
+	}
```

Comment:
probably need to pull this out into a shared stylesheet (probably need to pull a LOT of CSS out into shared stylesheets lol)

or, maybe the terms stuff should be wrapped up in a component? might be the responsible choice

Reply:
definitely not a bad idea. I haven't looked at Storybook + Astro